build/print.go

```diff
@@ -453,7 +453,11 @@ func (p *printer) expr(v Expr, outerPrec int) {
 		p.seq("{}", v.List, &v.End, modeList, false, v.ForceMultiLine)
 
 	case *TupleExpr:
-		p.seq("()", v.List, &v.End, modeTuple, v.ForceCompact, v.ForceMultiLine)
+		mode := modeTuple
+		if !v.Start.IsValid() {
+			mode = modeSeq
+		}
+		p.seq("()", v.List, &v.End, mode, v.ForceCompact, v.ForceMultiLine)
```

Comment:
thought: consistently unwrapping v.ForceCompact, v.ForceMultiLine is a lot of boilerplate that might introduce a bug sometime if we accidentally swap and pass (..., v.ForceMultiLine, v.ForceCompact) 

todo: add methods IsForceCompact(), IsForceMultiLine() to Expr and then pass an interface declaring those methods to p. seq ?

Reply:
I had a similar thought: to get rid of these attributes and calculate the mode in the printer by looking at `.Span()`s of sequences and their elements. This will make the parser and the AST simpler and in theory replaceable with the parser of Skylark in Go.